packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/transfer/database/database_stub.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3_dart/src/storage_s3_service/transfer/database/transfer_record.dart';
+
+/// {@template amplify_storage_s3_dart.transfer_database}
+/// Database to persist [TransferRecord] created by Storage S3 plugin upload
+/// operations.
+/// {@endtemplate}
+class TransferDatabase {
```

Comment:
Suggest making this `abstract` and removing method bodies.

```suggestion
abstract class TransferDatabase {
```

The only reason we would choose to add method bodies w/ `UnimplementedError` would be if there was a risk of breaking customers by adding APIs. In contrast, `abstract` w/out bodies makes it breaking when adding/removing APIs and forces implementations to implement all methods which I think we want here.

Reply:
I don't think we can use dependancy manager with abstract class

Reply:
Hmm it's a bit confusing since this class acts as both the interface and the default implementation. That alone is not a pattern I would suggest, but I suppose is fine for this PR.